pages/docs/concepts/asyncapi-document/tags.md

```diff
@@ -0,0 +1,249 @@
+```
+
+### `Tags` in `Operations` Object
+The `tags` object within the `operations` object of the AsyncAPI document allows for logical grouping and categorization of individual `operation` objects based on the type of operation or functionality and more. The `tags` object when used within a `operations` object, the context is restricted to the `operations` object and the individual `operation` of the AsyncAPI document, meaning they only impact the `operations` object of the AsyncAPI document. 
+
+Here's a visual representation of the `tags` object inside a `operations` object in an AsyncAPI document:
+```mermaid
+graph LR
+  A[operation]
+  B(title)
+  C(summary)
+  D(description)
+  E(channel)
+  F(action)
+  G(security)
+  H(tags)
+  I(bindings)
+  J(traits)
+  K(messages)
+  L(reply)
+  M(address)
+  N(channel)
+  W(asyncapi)
+  X(name)
+  Y(description)
+  Z(externalDocs)
+  
+  W --> A
+  A --> B
+  A --> C
+  A --> D
+  A --> E
+  A --> F
+  A --> G
+  A --> H
+  A --> I
+  A --> J
+  A --> K
+  A --> L
+  L --> M
+  L --> N
+  H --> X
+  H --> Y
+  H --> Z
+```
+
+Below is an example of the `tags` object inside the `operations` object in an AsyncAPI document:
+```yaml
+operations:
+  onUserSignUp:
+    title: User sign up
+    summary: Action to sign a user up.
+    description: A longer description
+    channel:
+      $ref: '#/channels/userSignup'
+    action: send
+    tags:
+      - name: user
+        description: operation related to user
+      - name: signup
+        description: operation related to a user's signUp
+      - name: register
+        description: operation related to a new registration
+    bindings:
+      amqp:
+        ack: false
+    traits:
+      - $ref: '#/components/operationTraits/kafka'
+```
+
+### `Tags` in `message` object
+Tags are associated with individual message objects allowing for logical grouping and categorization of messages based on specific requirements or criteria or given channels and operations and more. The `tags` object when used within a `message` object, the context is restricted to the `message` object of the AsyncAPI document, meaning they only impact the `message` object of the AsyncAPI document. 
+
+Here's a visual representation of a `tags` object inside a `message` object in an AsyncAPI document:
+```mermaid
+graph LR
+  A[message] --> B[summary]
+  A --> C[tags]
+  C --> D[name]
+  C --> E[description]
+  A --> F[headers]
+  F --> G[correlationId]
+  F --> H[applicationInstanceId]
+  A --> I[payload]
+```
+
+Below is an example of the `tags` object inside the `message` object in an AsyncAPI document:
+```yaml
+ name: SimpleSignup
+summary: A simple UserSignup example message
+tags: 
+    - name: userSignUp
+      description: some message related to user signup
+headers:
+  correlationId: my-correlation-id
+  applicationInstanceId: myInstanceId
+payload:
+  user:
+    someUserKey: someUserValue
+  signup:
+    someSignupKey: someSignupValue
+```
+
```

Comment:
the best would be to have at the end a summary example, with all the above examples combined into one AsyncAPI document, where all tags are defined in components, and then referenced in other components, you know what I mean?